File: cpp/205_IsomorphicStrings.cpp
class Solution {
public:
    bool isIsomorphic(string s, string t) {
        std::unordered_map<char,int> st_hash;
        std::unordered_map<char,int> ts_hash;
        
        for(int i = 0; i < s.size(); ++i){
            if(st_hash[s[i]] != ts_hash[t[i]]) return false;
            else st_hash[s[i]] = ts_hash[t[i]] = i+1;
        }
        return true;
    }
};

File: js/205_IsomorphicStrings.js
va/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isIsomorphic = function(s, t) {
    var st_hash = {};
    var ts_hash = {};
    for(var i in s){
        if(!st_hash[s[i]]) st_hash[s[i]] = i+1;
        if(!ts_hash[t[i]]) ts_hash[t[i]] = i+1;
        if(st_hash[s[i]] !== ts_hash[t[i]]) return false;
    }
    return true;
};